pkg/querier/blocks_consistency_checker.go

```diff
@@ -52,29 +56,33 @@ func (c *BlocksConsistencyChecker) Check(expectedBlocks []ulid.ULID, knownDeleti
 	missingBlocks := map[string][]string{}
 	var missingBlockIDs []string
 
-	for _, blockID := range expectedBlocks {
+	for _, meta := range expectedBlocks {
 		// Some recently uploaded blocks, already discovered by the querier, may not have been discovered
 		// and loaded by the store-gateway yet. In order to avoid false positives, we grant some time
 		// to the store-gateway to discover them. It's safe to exclude recently uploaded blocks because:
 		// - Blocks uploaded by ingesters: we will continue querying them from ingesters for a while (depends
 		//   on the configured retention period).
 		// - Blocks uploaded by compactor: the source blocks are marked for deletion but will continue to be
 		//   queried by store-gateways for a while (depends on the configured deletion marks delay).
-		if ulid.Now()-blockID.Time() < uint64(c.uploadGracePeriod.Milliseconds()) {
+		if c.uploadGracePeriod > 0 && time.Since(meta.UploadedAt) < c.uploadGracePeriod {
+			level.Debug(c.logger).Log("msg", "block skipped from consistency check because recently uploaded", "block", meta.ULID.String(), "uploadedAt", meta.UploadedAt.String())
 			continue
 		}
 
 		// The store-gateway may offload blocks before the querier. If that happens, the querier will run a consistency check
 		// on blocks that can't be queried because they were offloaded. For this reason, we don't run the consistency check on any block
 		// which has been marked for deletion more then "grace period" time ago. Basically, the grace period is the time
 		// we still expect a block marked for deletion to be still queried.
-		if mark := knownDeletionMarks[blockID]; mark != nil {
-			if time.Since(time.Unix(mark.DeletionTime, 0)) > c.deletionGracePeriod {
+		if mark := knownDeletionMarks[meta.ULID]; mark != nil {
+			deletionTime := time.Unix(mark.DeletionTime, 0)
+
+			if c.deletionGracePeriod > 0 && time.Since(deletionTime) > c.deletionGracePeriod {
+				level.Debug(c.logger).Log("msg", "block skipped from consistency check because marked for deletion", "block", meta.ULID.String(), "deletionTime", deletionTime.String())
```

Comment:
```suggestion
				level.Debug(c.logger).Log("msg", "block skipped from consistency check because it is marked for deletion", "block", meta.ULID.String(), "deletionTime", deletionTime.String())
```